107 Polymorphism commands

ikfir opened the pull request:
<!-- Thanks for taking the time to write this Pull Request! ❤️ -->

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [x] 📖 Documentation (updates to the documentation)
- [x] 🐞 Bug fix (fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)

### 📚 Description
When we have abstract class that has abstract method with `@SubCommand` annotation we will want it to run on the implmenetion

```java
public abstract class BaseCommand implements CommandNode {

    @SubCommand("example <int>")
    public abstract void example(Sender sender, int magicValue);

}

@Command("example")
public class BaseCommandImpl extends BaseCommand {

    @Override
    public void example(Sender sender, int magicValue) {
    // Do something
    }

}
```

Resolves #107 
<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Resolves #1337" -->

### 🧪 How Has This Been Tested?
`BaseCommandTest`
`UsingBaseCommandTest`
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

ikfir committed:
Add support for super class

Close #107

ikfir committed:
Add support for super class of super class and interfaces
also fixes GITHUB_WORKFLOW_REF null will just be snapshot

Close #107

ikfir committed:
I think super class is ready shhh nice name

resolves #107

liorsl commented on common/src/main/java/net/apartium/cocoabeans/reflect/ClassUtils.java:
```diff
@@ -0,0 +1,25 @@
+
+public class ClassUtils {
+
+    public static Collection<Class<?>> getSuperClassAndInterfaces(Class<?> clazz) {
```
Change to a method that gets all method instances from clazz (including inheritance), use reflection cache

ikfir replied:
how is that related to class utils?
We could add reflection cache for classes if that what you want

ikfir committed:
Merge branch 'main' of github.com:PoweredByApartium/cocoa-beans into 107-super-class-subcommand

ikfir committed:
Added tests and fixes ClassUtils

there are still an error with class utils

Resolves #107

ikfir committed:
Added tests and fixes ClassUtils & method utils

TODO:
  - Reflection cache

Resolves #107

ikfir committed:
Merge branch 'main' of github.com:PoweredByApartium/cocoa-beans into 107-super-class-subcommand

ikfir committed:
Security features for player parser

You can only find player who you can see

ikfir committed:
Start working on docs

Resolves #107

ikfir committed:
Added to examples Polymorphism

Resolves #107

ikfir committed:
Added source parser write more about polymor

also edited a lot of docs and make it look cleaner
add alot of todo for lior <3

Resolves #107

ikfir committed:
Finish the wiki and fixes ReflectionCache & MethodUtils

Closes #107

ikfir committed:
Added test to player parser hidden

Also added context to error it could be null as well btw
Closes #107
